### DISCUSSION TITLE
R formula: brew cleanup deletes the site-library folder if it is empty

### DISCUSSION THREAD

**Discussion by pooranis** [Homebrew/discussions#3239] [Everyday usage]
I noticed this in installing [homebrew-core R](https://github.com/Homebrew/homebrew-core/blob/master/Formula/r.rb) 4.2.0 (and earlier versions), but I am not sure if something is going wrong uniquely with me and my situation. 

1. As part of installation, the site-library folder gets created in _$HOMEBREW_PREFIX/lib/R/4.2/site-library_ if it doesn't exist (and then is symlinked to _$HOMEBREW_PREFIX/Cellar/r/4.2.0/lib/R/site-library_  so if you upgrade minor version of R can keep previously installed packages ).  
2. Then if you happen to run  `brew cleanup`, brew deletes it if it is empty (which if you haven't installed any R pkgs yet, it will be).  
    - **I'm not sure if this happens by design (?) or is a bug that this particular folder is not preserved or is the fault of something wrong on my end.**
4. The next time you want to install a package in R, depending on at what juncture the folder got deleted and your setup, you may get a warning/error that the _site-library_ directory doesn't exist AND/OR it will install to the system library folder _$HOMEBREW_PREFIX/Cellar/r/4.2.0/lib/R/library_ if you don't have a user library directory set up 
    - (suboptimal, as you will lose your packages on next minor update - :( happened to me :( ) .  

5. If at some point, you have to relink ``brew unlink r && brew link r`` because of the previous symlink made in step 1, a link gets created in _$HOMEBREW_PREFIX/lib/R/site-library_ - and it ends up just being a "circular link" - broken.
    - (I've done this many times because I never understood where that _site-library_ folder went and thought it was some symlink problem) 

So, now I manually created that _$HOMEBREW_PREFIX/lib/R/4.2/site-library_ folder and put a dummy file in it, so for the period it is empty and I am installing other things, it stays put, but it occurred to me that this shouldn't be happening.

**Comment by occludedpixel** [comment]
I think this is resolved by #4343 ?

I was also wondering why `brew cleanup` doesn't seem to delete the old site-library folders ?
